### PR TITLE
MariaDB config rebase and cleanup

### PR DESCRIPTION
**Pull request by rsalevsky**
Thinks to do:

- [ ] Slow query logging which works
- [ ] Move tuning related options to separate log and check if they still make sense

### CODE REVIEW

File: chef/cookbooks/mysql/recipes/server.rb [hunk 1]
```diff
@@ -77,18 +77,29 @@
...
-template "/etc/my.cnf" do
+template "/etc/my.cnf.d/crowbar.cnf" do
   source "my.cnf.erb"
   owner "root"
   group "mysql"
   mode "0640"
-  variables(
-    cluster_addresses: cluster_addresses
-  )
   notifies :run, resources(script: "handle mysql restart"), :immediately if platform_family?("debian")
```
Comment (jsuchome) [inline chef/cookbooks/mysql/recipes/server.rb]: As we're removing debian parts elsewhere, we could probably do here as well

**Comment by jsuchome** [comment]
[P4 (5) Sanitize and enhance mariadb configuration done by cookbook](https://trello.com/c/PcPTyEX7/1-p4-5-sanitize-and-enhance-mariadb-configuration-done-by-cookbook)

File: chef/cookbooks/mysql/attributes/server.rb [hunk 2]
```diff
@@ -18,63 +18,25 @@
 #
 
 default["mysql"]["bind_address"]              = node["ipaddress"]
-default["mysql"]["storage_engine"]            = "InnoDB"
-default["mysql"]["datadir"]                   = "/var/lib/mysql"
 default["mysql"]["tmpdir"]                    = "/var/lib/mysqltmp"
-default["mysql"]["logdir"]                    = "/var/lib/mysqllogs"
```
Comment (jsuchome) [inline chef/cookbooks/mysql/attributes/server.rb]: You also have to remove `logdir` directory creation from server.rb

Comment (rsalevsky) [inline chef/cookbooks/mysql/attributes/server.rb]: Done

**Comment by rsalevsky** [comment]
Not finished yet, but we can already merge this PR and I will create a followup PR where I do the fine tuning.

File: chef/cookbooks/mysql/recipes/server.rb (hunk 1 above)
Comment (nicolasbock) [inline chef/cookbooks/mysql/recipes/server.rb]: Same question: What about debian?

File: chef/cookbooks/mysql/templates/default/my.cnf.erb [hunk 3]
```diff
@@ -78,122 +19,22 @@ bind-address            = <%= node['mysql']['bind_address'] %>
 #
 # * Fine Tuning
 #
-key_buffer              = <%= node['mysql']['tunable']['key_buffer'] %>
+
+# Remove after MariaDB 10.2.X switch (new default is 16777216)
 max_allowed_packet      = <%= node['mysql']['tunable']['max_allowed_packet'] %>
-thread_stack            = <%= node['mysql']['tunable']['thread_stack'] %>
+# Remove after MariaDB 10.2.X switch (new default is auto)
```
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/mysql/templates/default/my.cnf.erb]: I'd prefer this was marked with FIXME or such

File: chef/cookbooks/mysql/templates/default/my.cnf.erb [hunk 4]
```diff
@@ -78,122 +19,22 @@ bind-address            = <%= node['mysql']['bind_address'] %>
...
-# For generating SSL certificates I recommend the OpenSSL GUI "tinyca".
-#
-# ssl-ca=/etc/mysql/cacert.pem
-# ssl-cert=/etc/mysql/server-cert.pem
-# ssl-key=/etc/mysql/server-key.pem
 
 [mysqldump]
-quick
-quote-names
+# Remove after MariaDB 10.2.X switch (new default is 16777216)
```
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/mysql/templates/default/my.cnf.erb]: As above.

File: chef/cookbooks/mysql/attributes/server.rb [hunk 5]
```diff
@@ -18,63 +18,23 @@
 #
 
 default["mysql"]["bind_address"]              = node["ipaddress"]
-default["mysql"]["storage_engine"]            = "InnoDB"
-default["mysql"]["datadir"]                   = "/var/lib/mysql"
```
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/mysql/attributes/server.rb]: I don't see any other code that reintroduces this, and this PR does not remove its use from crowbar_framework/app/views/barclamp/database/_edit_attributes.html.haml and friends, so it's at best a no-op, and at worst an error. Do we intend to keep it configurable?

Comment (jsuchome) [inline chef/cookbooks/mysql/attributes/server.rb]: Oh, true, I believe we still want this

File: chef/cookbooks/mysql/templates/default/galera.cnf.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+wsrep_provider=/usr/lib64/galera-3/libgalera_smm.so
```
Comment (jsuchome) [inline chef/cookbooks/mysql/templates/default/galera.cnf.erb]: Seen this in one of @jhesketh commits - do we need to specify correct section of this part? (That would be [mysqld] I believe)

Comment (jhesketh) [inline chef/cookbooks/mysql/templates/default/galera.cnf.erb]: I believe we need to, yes.

Comment (rsalevsky) [inline chef/cookbooks/mysql/templates/default/galera.cnf.erb]: Fixed

File: chef/cookbooks/mysql/metadata.rb [hunk 7]
```diff
@@ -26,11 +26,6 @@
           description: "Address that mysqld should listen on",
           default: "ipaddress"
 
-attribute "mysql/datadir",
```
Comment (jsuchome) [inline chef/cookbooks/mysql/metadata.rb]: This should not go away

File: chef/cookbooks/mysql/templates/default/my.cnf.erb [hunk 8]
```diff
@@ -2,74 +2,15 @@
...
-#   If you make changes to these settings and your system uses apparmor, you may
-#   also need to also adjust /etc/apparmor.d/usr.sbin.mysqld.
-#
-
 user                   = mysql
-pid-file               = <%= node['mysql']['pid_file'] %>
-socket                 = <%= node['mysql']['socket'] %>
-port                   = 3306
-basedir                = /usr
-datadir                = <%= node['mysql']['datadir'] %>
```
Comment (jsuchome) [inline chef/cookbooks/mysql/templates/default/my.cnf.erb]: Should not be dropped

File: chef/cookbooks/mysql/recipes/server.rb [hunk 9]
```diff
@@ -77,16 +70,25 @@
 nodes_names = cluster_nodes.map { |n| n[:hostname] }
 cluster_addresses = "gcomm://" + nodes_names.join(",")
 
-template "/etc/my.cnf" do
+template "/etc/my.cnf.d/crowbar.cnf" do
```
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/mysql/recipes/server.rb]: server.cnf ? crowbar.cnf might confuse others as to if this about the crowbar database, which it isn't.

Comment (rsalevsky) [inline chef/cookbooks/mysql/recipes/server.rb]: I will split the config further in my next PR and of course will give it a proper name.

**Pull request merged**